janusgraph-codepipelines-ci/README.md

```diff
@@ -0,0 +1,124 @@
+relationship and replace the principal `ec2.amazonaws.com` with `codepipeline.amazonaws.com`.
+
+## Setup environment
+Follow the steps below to prepare to create a test pipeline and run tests. First, clean up and set some constant
+environment variables.
+
+```bash
+mvn clean package
+#constants
+export AWS_PROFILE_NAME='code-pipelines'
+export AWS_ACCOUNT_NUMBER=`aws sts get-caller-identity --profile ${AWS_PROFILE_NAME} --output text | cut -f1`
+```
+
+Second, configure the parameters below to customize your test stack.
+
+```bash
+#GitHub personal access token you created in step 1
+export GITHUB_TOKEN=''
+#GitHub organization or user name of the repository to build
+export GITHUB_USERNAME=''
+#GitHub repository name to use for builds
+export GITHUB_REPOSITORY=''
+#GitHub branch in the above repository to use for builds
+export GITHUB_BRANCH=''
+#Name of the AWS CodeBuild service role you created in step 4
+export AWS_CODEBUILD_ROLE=''
+#Name of the AWS CodePipeline role you created in step 6
+export AWS_CODEPIPELINE_ROLE=''
+#Region you want to set up your test stack in
+export AWS_REGION_NAME=''
+#Name of bucket you want to use or create and use for storing your build artifacts.
+export AWS_S3_BUCKET_NAME=''
+#Name of pipeline JSON configuration file to use
+export PIPELINE_CONFIGURATION=''
+```
+
+## Create test pipeline
+This tool uses templates defined in JSON files to configure parallel builds in AWS CodePipeline. By default,
+you can have up to [five parallel actions per stage](http://docs.aws.amazon.com/codepipeline/latest/userguide/limits.html)
+in a pipeline and up to [twenty parallel builds](http://docs.aws.amazon.com/codebuild/latest/userguide/limits.html#limits-builds)
+per region per account. If you need more, you can
+[request to be whitelisted](http://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html) for more.
+
+Here is an example of a file that defines two pipelines with parallel build actions.
+
+```json
+{
+    "j1" : {
+        "bdb" : { "MODULE" : "janusgraph-berkeleyje" },
+        "h-l-s" : { "MODULE" : "janusgraph-hadoop-parent/janusgraph-hadoop-2,janusgraph-lucene,janusgraph-solr" },
+        "cassandra" : { "MODULE" : "janusgraph-cassandra" },
+        "test" : { "MODULE" : "janusgraph-test" }
+    },
+    "j2" : {
+        "hbase098" : { "MODULE" : "janusgraph-hbase-parent/janusgraph-hbase-098" },
+        "hbase10" : { "MODULE" : "janusgraph-hbase-parent/janusgraph-hbase-10" },
+        "es" : { "MODULE" : "janusgraph-es" },
+        "cql" : { "MODULE" : "janusgraph-cql" }
+    }
+}
+```
+The first pipeline is called `j1` and the second pipeline is called `j2`. Each of these pipelines have four
+parallel build actions defined. Each build action is keyed at the action name and includes the environment
+variables to be passed CodeBuild and used in the buildspec.yml file. Use `export PIPELINE_CONFIGURATION=pipe.json` for regular tests
```

Comment:
Instead of "Use 'export PIPELINE..." how about something like below?

Define the `PIPELINE_CONFIGURATION` environment variable to point to the custom pipeline file or use existing pipe.json/tp-pipe.json to run all default/TinkerPop tests.

```bash
export PIPELINE_CONFIGURATION=pipe.json
```